File: app/src/main/java/com/smarttoolfactory/toolbarsamples/activity/Activity6ParallaxTransparentImageTabs.java
package com.smarttoolfactory.toolbarsamples.activity;

import com.google.android.material.tabs.TabLayout;
import com.smarttoolfactory.toolbarsamples.R;

import android.os.Bundle;

import com.google.android.material.appbar.CollapsingToolbarLayout;
import com.smarttoolfactory.toolbarsamples.fragment.OneFragment;
import com.smarttoolfactory.toolbarsamples.fragment.ThreeFragment;
import com.smarttoolfactory.toolbarsamples.fragment.TwoFragment;
import com.smarttoolfactory.toolbarsamples.fragment.ViewPagerAdapter;


import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.viewpager.widget.ViewPager;

import android.view.MenuItem;

public class Activity6ParallaxTransparentImageTabs extends AppCompatActivity {

    private Toolbar toolbar;
    private TabLayout tabLayout;
    private ViewPager viewPager;
    ViewPagerAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.appbar6_parallax_image_tabs);

        toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        viewPager = (ViewPager) findViewById(R.id.viewPager);
        setupViewPager(viewPager);
        tabLayout = (TabLayout) findViewById(R.id.tablayout);
        tabLayout.setupWithViewPager(viewPager);

        final CollapsingToolbarLayout collapsingToolbarLayout = (CollapsingToolbarLayout) findViewById(
                R.id.collapsingToolbarLayout);
        collapsingToolbarLayout.setTitleEnabled(false);


        viewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {

            @Override
            public void onPageSelected(int arg0) {
                adapter.notifyDataSetChanged();
                // System.out.println("Viewpager onPageSelected: " + arg0);
            }

            @Override
            public void onPageScrolled(int arg0, float arg1, int arg2) {
            }

            @Override
            public void onPageScrollStateChanged(int arg0) {

            }
        });
    }

    private void setupViewPager(ViewPager viewPager) {
        adapter = new ViewPagerAdapter(getSupportFragmentManager());
        adapter.addFragment(new OneFragment(), "FIRST");
        adapter.addFragment(new TwoFragment(), "SECOND");
        adapter.addFragment(new ThreeFragment(), "THIRD");
        viewPager.setAdapter(adapter);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                return true;
            case R.id.action_settings:
                return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/smarttoolfactory/toolbarsamples/activity/Activity6ParallaxTransparentImageTabs.kt
package com.smarttoolfactory.toolbarsamples.activity

import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.tabs.TabLayout
import android.os.Bundle
import android.view.Menu
import com.smarttoolfactory.toolbarsamples.R
import android.view.View
import com.google.android.material.appbar.CollapsingToolbarLayout
import android.view.MenuItem
import androidx.appcompat.widget.Toolbar
import androidx.viewpager2.widget.ViewPager2
import com.google.android.material.tabs.TabLayoutMediator
import com.smarttoolfactory.toolbarsamples.adapter.PostActvitiyStateAdapter

class Activity6ParallaxTransparentImageTabs : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity6_parallax_image_tabs)

      val  toolbar = findViewById<View>(R.id.toolbar) as Toolbar
        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        val viewPager2 = findViewById<ViewPager2>(R.id.viewPager2)

        viewPager2.adapter = PostActvitiyStateAdapter(this)

        val tabLayout = findViewById<TabLayout>(R.id.tabLayout)

        TabLayoutMediator(tabLayout, viewPager2) { tab, position ->
            tab.text = "Tab $position"
        }.attach()

        val collapsingToolbarLayout = findViewById<View>(
            R.id.collapsingToolbar
        ) as CollapsingToolbarLayout

        collapsingToolbarLayout.isTitleEnabled = false

    }



    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.main, menu)
        // return true so that the menu pop up is opened
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}